### PR TITLE
Follow-up to PR #2776

### PR DESCRIPTION
**Pull request by luckysori**
Tackling most of @thomaseizinger's suggestions from his review of PR #2776.

Patch af9643e could either be squashed with 2a44449 or reverted and replaced with the introduction of another macro for spawning swaps. This depends on your reviews.

### CODE REVIEW

File: cnd/src/db/errors.rs [hunk 1]
```diff
@@ -33,4 +33,4 @@ pub struct NoHbitRefundIdentity(pub LocalSwapId);
 
 #[derive(thiserror::Error, Debug, Clone, Copy)]
 #[error("no hbit redeem identity found in database for swap {0}")]
-pub struct NoHbit20RedeemIdentity(pub LocalSwapId);
```
Comment (thomaseizinger) [inline cnd/src/db/errors.rs:36]: LOL :joy:

File: cnd/src/with_swap_types.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
+#[macro_export]
+macro_rules! with_swap_types {
```
Comment (thomaseizinger) [inline cnd/src/with_swap_types.rs:2]:
```suggestion
macro_rules! within_swap_context {
```

? :)

File: cnd/src/storage.rs [hunk 3]
```diff
@@ -1039,57 +1045,6 @@ impl
...
-            #[sql_type = "sql_types::Text"]
-            role: Text<Role>,
-            #[sql_type = "sql_types::Text"]
-            alpha_protocol: Text<Protocol>,
-            #[sql_type = "sql_types::Text"]
-            beta_protocol: Text<Protocol>,
-        }
-
-        let Result { role, alpha_protocol, beta_protocol } = self.db.do_in_transaction(|connection| {
-            // Here is how this works:
```
Comment (thomaseizinger) [inline cnd/src/storage.rs:1068]: Is this comment preserved in the other impl that we are now using?

Comment (luckysori) [reply cnd/src/storage.rs:1068]: Good question! It is :)

File: cnd/src/with_swap_types.rs (hunk 2 above)
Comment (luckysori) [reply cnd/src/with_swap_types.rs:2]: I like that!

**Comment by tcharding** [comment]
Poo, there is a bunch of changes to `storage.rs` that conflict with the work I did over the last two days on https://github.com/comit-network/comit-rs/issues/2768. I'll wait for this to go in and then rebase on top. Some changes seem arbitrary e.g. you favoured using `db::tables::SecretHash` and brought `comit::SecretHash` into scope I did it the other way round, neither seems overly better than the other as long as we are uniform. I'l attempt to be so when rebaseing.

**Comment by luckysori** [comment]
> Can I offer one suggestion; if you have 'random' changes within a patch it makes the review harder because the reviewer has to think 'why has that changed, it has nothing to do with what the commit log said this patch does'. For example, there is an impl block removed in one of the patches. If you want to do this while you are working just throw the change into a separate patch, makes review super easy then. Cheers.

I can certainly do that! I like to strike a balance between atomic commits and too many commits though. I'll keep it in mind nonetheless :)

**Comment by luckysori** [comment]
> Poo, there is a bunch of changes to `storage.rs` that conflict with the work I did over the last two days on #2768. I'll wait for this to go in and then rebase on top. Some changes seem arbitrary e.g. you favoured using `db::tables::SecretHash` and brought `comit::SecretHash` into scope I did it the other way round, neither seems overly better than the other as long as we are uniform. I'l attempt to be so when rebaseing.

Uniformity was also my goal with some of those unrelated changes in certain patches, so I'm all for it!

**Pull request merged**